### PR TITLE
Chore: upgrade eslint

### PR DESCRIPTION
**Pull request by philipyoo**
I tried to separate my commits by each type of eslint change, so it might be easiest to view by each commit.

Let me know what you think, if we should add or remove certain rules. Current rules can be viewed:
https://github.com/scality/Guidelines/blob/chore/upgradeEslint/index.js

List of Eslint rules: https://eslint.org/docs/rules/

Note: Indent fixes were mainly done by the linter `--fix` command

Note: Will squash commits after making any changes based on peoples' feedback

### CODE REVIEW

File: lib/s3routes/routes/routeGET.js [hunk 1]
```diff
@@ -74,6 +74,7 @@ function routerGET(request, response, api, log, statsClient, dataRetrievalFn) {
                 });
         }
     } else {
+        /*eslint-disable no-lonely-if*/
```
Comment (bennettbuchanan) [inline lib/s3routes/routes/routeGET.js]: Why disable it here?

File: lib/storage/utils.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ const assert = require('assert');
 
 function trySetDirSyncFlag(path) {
     // may throw if ioctl is not available
+    // eslint-disable-next-line
```
Comment (bennettbuchanan) [inline lib/storage/utils.js]: Can drop this rule.

File: lib/s3routes/routes/routeGET.js (hunk 1 above)
Comment (philipyoo) [reply lib/s3routes/routes/routeGET.js]: I wasn't sure how to refactor this part because each nested if seemed like it served a purpose :/

**Comment by philipyoo** [comment]
@ironman-machine try plz

File: lib/auth/in_memory/Indexer.js [hunk 3]
```diff
@@ -126,7 +126,8 @@ class Indexer {
      * @return {Object} entity.email - The entity's lowercased email
      */
     getEntityByKey(key) {
-        if (this.accountsBy.accessKey.hasOwnProperty(key)) {
+        if (Object.prototype.hasOwnProperty.call(
```
Comment (rahulreddy) [inline lib/auth/in_memory/Indexer.js]: I think at this point, it's just easy to call 
```js
if (this.accountsBy.accessKey[key] !== undefined) {
```

**Comment by philipyoo** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine r+

**Pull request merged**